helpers/pathutils/merge.go

```diff
@@ -0,0 +1,30 @@
+package pathutils
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/path"
+)
+
+// MergeExpressionsWithAttribute returns the given path.Expressions,
+// but each has been merged with the given attribute path.Expression,
+// and then resolved.
+//
+// Additionally, if the attribute path.Expression was not part of the initial slice,
+// it is added to the result.
+func MergeExpressionsWithAttribute(pathExps path.Expressions, attrPathExp path.Expression) path.Expressions {
+	result := make(path.Expressions, 0, len(pathExps)+1)
+
+	// First, add the attribute own path expression to the result
+	result = append(result, attrPathExp)
+
+	for _, pe := range pathExps {
+		mpe := attrPathExp.Merge(pe).Resolve()
+
+		// Include the merged path expression,
+		// only if it's not the same as the attribute
+		if !mpe.Equal(attrPathExp) {
+			result = append(result, mpe)
+		}
+	}
+
+	return result
+}
```

Comment:
I'd be a fan of a `(path.Expressions).Append(path.Expression)`.

Ah, and thanks to catch the unnecessary call to `.Resolve()`.

Reply:
Given this will be internal use, I'm happy to go back and use `.Append()` at later stage once it's available.

Reply:
Sorry to pick at words, but you mention "this will be internal use", yet this package and function are publicly exported rather than in an `internal/` package. If it really should be hidden from the public API of the module, it'll need to be moved into an `internal/` directory so only this module has access to it. Hopefully https://github.com/hashicorp/terraform-plugin-framework/pull/401 can eliminate the need for this altogether though. 😄